aiidalab_widgets_base/viewers.py

```diff
@@ -34,32 +34,22 @@ def registration_decorator(widget):
     return registration_decorator
 
 
-def viewer(obj, downloadable=True, **kwargs):
+def viewer(obj, **kwargs):
     """Display AiiDA data types in Jupyter notebooks.
 
-    :param downloadable: If True, add link/button to download the content of displayed AiiDA object.
-    :type downloadable: bool
-
     Returns the object itself if the viewer wasn't found."""
     if not isinstance(obj, orm.Node):  # only working with AiiDA nodes
         warnings.warn(
             f"This viewer works only with AiiDA objects, got {type(obj)}", stacklevel=2
         )
         return obj
 
-    try:
+    if obj.node_type in AIIDA_VIEWER_MAPPING:
         _viewer = AIIDA_VIEWER_MAPPING[obj.node_type]
-    except KeyError as exc:
-        if obj.node_type in str(exc):
-            warnings.warn(
-                f"Did not find an appropriate viewer for the {type(obj)} object. Returning the object "
```

Comment:
I don't find this warning particularly helpful since it is kinda self-evident. Also for Base types like `Int` it's perfectly okay not to have a dedicated viewer.